pop-api/examples/filtered-call/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+/// This is an example contract used for unit testing purposes.
+/// It is intended to ensure that the call filter for the Pop API chain
+/// extension works as expected.
+/// An Ok(()) returned from `get_filtered` means that the filter successfully
+/// BLOCKED the call. An Err() means that the filter did not block the call,
+/// which is incorrect behavior.
+/// The pop-api crate is not used because it does not support invalid calls ;)
+use scale::Encode;
+
+use ink::{env::Environment, prelude::vec::Vec};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum ContractError {
+	CallWasNotFiltered,
+}
+
+#[derive(Encode)]
+pub enum SystemCalls {
+	#[codec(index = 0)]
+	Remark { remark: Vec<u8> },
+}
+#[derive(scale::Encode)]
+pub enum RuntimeCall {
+	#[codec(index = 0)]
+	System(SystemCalls),
+}
+
+pub type Result<T> = core::result::Result<T, ContractError>;
+#[ink::chain_extension(extension = 909)]
+pub trait PopApi {
```

Comment:
This might be confusing imo, suggest renaming to something like `MyPopApi` to differentiate